ArduCopter/mode_acro.cpp

```diff
@@ -16,27 +16,35 @@ void Copter::ModeAcro::run()
 
     if (!motors->armed()) {
         // Motors should be Stopped
-        motors->set_desired_spool_state(AP_Motors::DESIRED_SHUT_DOWN);
+        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::SHUT_DOWN);
     } else if (ap.throttle_zero) {
         // Attempting to Land
-        motors->set_desired_spool_state(AP_Motors::DESIRED_GROUND_IDLE);
+        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::GROUND_IDLE);
     } else {
-        motors->set_desired_spool_state(AP_Motors::DESIRED_THROTTLE_UNLIMITED);
+        motors->set_desired_spool_state(AP_Motors::DesiredSpoolState::THROTTLE_UNLIMITED);
     }
 
-    if (motors->get_spool_mode() == AP_Motors::SHUT_DOWN) {
+    switch (motors->get_spool_state()) {
+    case AP_Motors::SpoolState::SHUT_DOWN:
         // Motors Stopped
         attitude_control->set_attitude_target_to_current_attitude();
         attitude_control->reset_rate_controller_I_terms();
-    } else if (motors->get_spool_mode() == AP_Motors::GROUND_IDLE) {
+        break;
+    case AP_Motors::SpoolState::GROUND_IDLE:
         // Landed
         attitude_control->set_attitude_target_to_current_attitude();
         attitude_control->reset_rate_controller_I_terms();
-    } else if (motors->get_spool_mode() == AP_Motors::THROTTLE_UNLIMITED) {
+        break;
+    case AP_Motors::SpoolState::THROTTLE_UNLIMITED:
         // clear landing flag above zero throttle
         if (!motors->limit.throttle_lower) {
             set_land_complete(false);
         }
+        break;
+    case AP_Motors::SpoolState::SPOOLING_UP:
+    case AP_Motors::SpoolState::SPOOLING_DOWN:
+        // do nothing and hope for the best!
```

Comment:
maybe we can shorten this comment to just "// do nothing".  I think we don't need to do anything during spool up or spool down.

Reply:
That makes me very curious, actually, and the reason I changed things to be switch statements.  In these states you should be on the ground - but we're not resetting targets and the like.

Needs attention from a controls person - if only to add a comment.

In the meantime I'll shorten it to "do nothing"

Reply:
Shortened all of these to just "do nothing"